CommonApplication.cmake

```diff
@@ -122,6 +123,34 @@ function(common_application Name)
     endif()
   endif()
 
+  if(NOT THIS_GUI OR THIS_DOXYGEN)
+    # run binary with --help to capture output for doxygen
+    set(_doc "${PROJECT_BINARY_DIR}/help/${Name}.md")
+    set(_cmake "${CMAKE_CURRENT_BINARY_DIR}/${Name}.cmake")
+    file(WRITE ${_cmake} "
+      execute_process(COMMAND \${APP} --help
+      OUTPUT_FILE ${_doc} ERROR_FILE ${_doc}.error)
+      file(READ ${_doc} _doc_content)
+      if(NOT _doc_content)
+        message(FATAL_ERROR \"${Name} is missing --help\")
+      endif()
+      file(WRITE ${_doc} \"${Name} {#${Name}}
+============
+
+```
+\${_doc_content}
+```
+\")
+")
+    add_custom_command(OUTPUT ${_doc}
+      COMMAND ${CMAKE_COMMAND} -DAPP=$<TARGET_FILE:${Name}> -P ${_cmake}
+      DEPENDS ${Name}
+      COMMENT "Creating help for ${Name}")
+    add_custom_target(${Name}-help ALL DEPENDS ${_doc})
+    add_dependencies(${PROJECT_NAME}-all ${Name}-help)
```

Comment:
Ok, will attach to project-doxygen.

For the record, I still think this is a bad idea since it makes the dependency chain even more entangled. As @tribal-tec shows, it's already not well understood:

* all is an internal, inaccessible CMake target
* project-all is our target to which all project dependencies are attached, now also used by people during dev work
* project-install depends on project-all (for obvious reasons)
* project-doxygen depends on project-install since it works on the installed headers only
* new: project-doxygen depends also on all project-help-targets